File: 02-Patterns/pattern08/solution.js
/*  

    * * * * * * * * *
      * * * * * * *
        * * * * * 
          * * *
            *     

*/

function pattern8(n) {
  for (let i = 0; i < 5; i++) {
    // left spaces
    for (let j = 0; j < i; j++) {
      process.stdout.write(" ");
    }

    // stars
    for (let j = 0; j < 2 * (n - i) - 1; j++) {
      process.stdout.write("*");
    }

    // right spaces
    for (let j = 0; j < i; j++) {
      process.stdout.write(" ");
    }

    console.log("");
  }
}

pattern8(5);

File: 02-Patterns/pattern08/solution.py
"""

    * * * * * * * * *
      * * * * * * *
        * * * * * 
          * * *
            *     

"""


def pattern8(n):
    for i in range(n):
        # left spaces
        for j in range(i):
            print("", end=" ")

        # stars
        for j in range(2 * (n - i) - 1):
            print("*", end="")

        # right spaces
        for j in range(i):
            print("", end=" ")

        print("")


pattern8(5)